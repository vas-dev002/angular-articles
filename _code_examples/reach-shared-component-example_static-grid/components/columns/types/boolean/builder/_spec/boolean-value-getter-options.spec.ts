import { ValueGetterParams } from "@ag-grid-community/core";
import { BooleanValueGetterOptions } from "..";

declare const expect: jest.Expect;

describe("[Builder option]: BooleanValueGetterOptions", () => {
	const columnName = "booleanImageColumn";
	let booleanValueGetterOptions: BooleanValueGetterOptions;

	beforeEach(() => {
		booleanValueGetterOptions = new BooleanValueGetterOptions(columnName);
	});

	describe("[BooleanValueGetterOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(booleanValueGetterOptions).toBeTruthy();
			expect(booleanValueGetterOptions).toMatchSnapshot();
		});
	});

	describe("[Method]: getterFunction", () => {
		it("should return the currect value if data is set", () => {
			const params = {
				data: { booleanImageColumn: "True" }
			};
			const expectedResult = params.data.booleanImageColumn;
			const result = booleanValueGetterOptions.getterFunction(params as unknown as ValueGetterParams);
			expect(result).toEqual(expectedResult);
		});

		it("should return null", () => {
			const params = { data: null };

			const result = booleanValueGetterOptions.getterFunction(params as unknown as ValueGetterParams);
			expect(result).toBe("");
		});
	});
});
