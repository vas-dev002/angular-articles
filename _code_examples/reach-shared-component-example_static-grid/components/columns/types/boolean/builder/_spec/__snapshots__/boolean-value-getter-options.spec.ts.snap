// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Builder option]: BooleanValueGetterOptions [BooleanValueGetterOptions instantiate] should be successfully created 1`] = `
BooleanValueGetterOptions {
  "columnName": "booleanImageColumn",
  "getterFunction": [Function],
}
`;
